src/ShareToken.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import firebase from './lib/firebase';
+import { db } from './lib/firebase';
+import { useCollection } from 'react-firebase-hooks/firestore';
+import { useHistory } from 'react-router-dom';
+import { withSnackbar } from 'notistack';
+
+const ShareToken = (props) => {
+  const [shareToken, setShareToken] = useState(null);
+
+  // 1. Make a on submit function where we pull list of collections
+  let history = useHistory();
+  const onSubmitHandler = (event) => {
+    event.preventDefault();
+    db.collection(shareToken)
+      .get()
+      .then((querySnapshot) => {
+        if (querySnapshot.empty) {
+          props.enqueueSnackbar('Incorrect token, please try again', {
+            variant: 'error',
+          });
+        } else {
+          props.enqueueSnackbar('Success', {
+            variant: 'success',
+          });
+          localStorage.setItem('userToken', shareToken);
+          history.push('/list');
+        }
+      });
+  };
+
+  // 2. Create change function that keeps track of user input
+  const onChangeHander = (event) => {
+    setShareToken(event.target.value);
+  };
+
+  // 3. In submit function we will compare the collections list with submitted token
+  // 4a. If exists then take user to shopping list
+  // 4b. If doesn't exist then notify user as such
+
+  return (
+    <>
+      <p>Join an existing shopping list by entering a three word token.</p>
```

Comment:
One thing we'll need to add later is to display the user's token somewhere so they can share it. It could be nice to use something like [React-copy-to-clipboard](https://www.npmjs.com/package/react-copy-to-clipboard) for ease of use. Unless I'm missing it, I don't think this is explicitly mentioned in any of the existing issues, so it might be something to add in the last weeks as we're focusing more on UI/UX. I'll make a note on our GitHub Discussions page! :)